framework/src/media/audio/audio_manager.c

```diff
@@ -1081,29 +1097,29 @@ int get_output_audio_volume(void)
 
 	return (int)g_audio_out_cards[g_actual_audio_out_card_id].config.volume;
 #endif
-	return 0;
+	return AUDIO_MANAGER_VOLUME_NOT_SUPPORT;
 }
 
 audio_manager_result_t set_input_audio_volume(uint8_t volume)
 {
-#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
+#ifndef CONFIG_AUDIO_EXCLUDE_MIC_GAIN
```

Comment:
As per I explained, f/w using h/w's configuration is not good option.
we already have a ioctl mechanism, we can use it.
Actually get_hardware_params gathers volume info through the get_audio_volume, definately audio_config_s's value is AUDIO_MANAGER_NOT_SUPPORT as a volume & gain

Reply:
All the exclude configs are removed in audio_manager.c